### ISSUE TITLE
Fix test placement of test_get_default_selection_1_a function

### ISSUE THREAD

**Issue by robertbartel** [NOAA-OWP/DMOD#271]
# TL;DR

* _dmod.core_ does not depend on _dmod.communication_
* _python/lib/core/dmod/test/test_allocationparadigm.py_ (being a unit test) should not import from _dmod.communication_
* right now, it does: _SchedulerRequestMessage_ 
* this problem import is needed for `test_get_default_selection_1_a`

# Details

The `test_get_default_selection_1_a` function is currently in _python/lib/core/dmod/test/test_allocationparadigm.py_.  The function compares the output of `AllocationParadigm.get_default_selection()` to `SchedulerRequestMessage.default_allocation_paradigm_str()` (both class methods) to make sure the two agree.   This made sense when the function was written, as _AllocationParadigm_ was in a different package that depended on _dmod.communication_.  

 _AllocationParadigm_ was moved to _dmod.core_, which does not depend on other internal packages.  As a result, the aforementioned **unit** test module now imports something the package itself isn't guaranteed to bring in.  

The test should somehow be updated to avoid this.  Initial ideas would be to move to _dmod.communication_ or to make this an integration test.

**Comment by robertbartel** [comment]
Putting at medium priority only because I have seen an issue with some other work that is related to this test function (though perhaps only circumstantially).
